crates/noirc_evaluator/src/lib.rs

```diff
@@ -304,7 +304,7 @@ impl Evaluator {
         // u8 and arrays are assumed to be private
         // This is not a short-coming of the ABI, but of the grammar
         // The new grammar has been conceived, and will be implemented.
-        let main = ir_gen.program.main();
+        let main = ir_gen.program.main_mut();
```

Comment:
Why do you now need main_mut ?

Reply:
`Program::main` returned a mutable reference to main previously, and I've split it into main and main_mut now. This reference was previously mutable so this instance was changed to main_mut to preserve semantics.

The two were split because the new ssa-gen pass does not mutate the monomorphized Program, and instead stores it in the `&'ssa SharedContext` struct, so it cannot use any mutable methods on it.

Reply:
Ah for some reason, I mis-read that change as adding main_mut where it was previously main_ref